src/global_scheduler/global_scheduler.cc

```diff
@@ -14,6 +14,45 @@
 #include "state/task_table.h"
 
 /**
+ * Retry the task assignment. If the local scheduler that the task is assigned
+ * to is no longer active, do not retry the assignment.
+ * TODO(rkn): We currently only retry the method if the global scheduler
+ * publishes a task to a local scheduler before the local scheduler has
+ * subscribed to the channel. If we enforce that ordering, we can remove this
+ * retry method.
+ *
+ * @param id The task ID.
+ * @param user_context The global scheduler state.
+ * @param user_data The Task that failed to be assigned.
+ * @return Void.
+ */
+void assign_task_to_local_scheduler_retry(UniqueID id,
+                                          void *user_context,
+                                          void *user_data) {
+  GlobalSchedulerState *state = (GlobalSchedulerState *) user_context;
+  Task *task = (Task *) user_data;
+  CHECK(Task_state(task) == TASK_STATUS_SCHEDULED);
+
+  // If the local scheduler has died since we requested the task assignment, do
+  // not retry again.
+  DBClientID local_scheduler_id = Task_local_scheduler(task);
+  auto it = state->local_schedulers.find(local_scheduler_id);
+  if (it == state->local_schedulers.end()) {
+    return;
```

Comment:
Let's avoid adding extra return statements and put everything below in an if statement

```C++
if (it != state->local_schedulers.end()) {
}
```

Reply:
Sorry for nitpicking, but why is that? I actually prefer the other way when reading code (put all short-circuit checks first, then do the actual function logic).

Reply:
Ok that's fine, I remember someone saying that code is easier to understand when functions only have one return statement.